pkg/server/commands/batch_check_command_test.go

```diff
@@ -0,0 +1,230 @@
+		mockCheckResolver.EXPECT().ResolveCheck(gomock.Any(), gomock.Any()).
+			Times(numChecks).
+			Return(nil, nil)
+
+		params := &BatchCheckCommandParams{
+			AuthorizationModelID: ts.GetAuthorizationModelID(),
+			Checks:               checks,
+			StoreID:              ulid.Make().String(),
+		}
+
+		result, _, err := cmd.Execute(context.Background(), params)
+		require.NoError(t, err)
+
+		// Quantity of correlation IDs should be equal
+		require.Equal(t, len(result), len(ids))
+
+		// And each ID should appear in the response
+		for _, id := range ids {
+			_, ok := result[CorrelationID(id)]
+			require.True(t, ok)
+		}
+	})
+
+	t.Run("fails_with_validation_error_if_too_many_tuples", func(t *testing.T) {
+		numChecks := int(maxChecks) + 1
+		checks := make([]*openfgav1.BatchCheckItem, numChecks)
+		for i := 0; i < numChecks; i++ {
+			checks[i] = &openfgav1.BatchCheckItem{
+				TupleKey: &openfgav1.CheckRequestTupleKey{
+					Object:   "doc:doc1",
+					Relation: "viewer",
+					User:     "user:justin",
+				},
+				CorrelationId: fmt.Sprintf("fakeid%d", i),
+			}
+		}
+
+		params := &BatchCheckCommandParams{
+			AuthorizationModelID: ts.GetAuthorizationModelID(),
+			Checks:               checks,
+			StoreID:              ulid.Make().String(),
+		}
+
+		_, _, err := cmd.Execute(context.Background(), params)
+
+		var expectedErr *BatchCheckValidationError
+		require.ErrorAs(t, err, &expectedErr)
+	})
+
+	t.Run("fails_with_validation_error_if_duplicated_correlation_ids", func(t *testing.T) {
+		numChecks := 2
+		checks := make([]*openfgav1.BatchCheckItem, numChecks)
+		for i := 0; i < numChecks; i++ {
+			checks[i] = &openfgav1.BatchCheckItem{
+				TupleKey: &openfgav1.CheckRequestTupleKey{
+					Object:   "doc:doc1",
+					Relation: "viewer",
+					User:     "user:justin",
+				},
+				CorrelationId: "hardcoded_id",
+			}
+		}
+
+		params := &BatchCheckCommandParams{
+			AuthorizationModelID: ts.GetAuthorizationModelID(),
+			Checks:               checks,
+			StoreID:              ulid.Make().String(),
+		}
+
+		_, _, err := cmd.Execute(context.Background(), params)
+		require.ErrorContains(t, err, "hardcoded_id")
+
+		var expectedErr *BatchCheckValidationError
+		require.ErrorAs(t, err, &expectedErr)
+	})
+
+	t.Run("fails_with_validation_error_if_empty_correlation_id", func(t *testing.T) {
+		numChecks := int(maxChecks) + 1
+		checks := make([]*openfgav1.BatchCheckItem, numChecks)
+		for i := 0; i < numChecks; i++ {
+			checks[i] = &openfgav1.BatchCheckItem{
+				TupleKey: &openfgav1.CheckRequestTupleKey{
+					Object:   "doc:doc1",
+					Relation: "viewer",
+					User:     "user:justin",
+				},
+				CorrelationId: fmt.Sprintf("fakeid%d", i),
+			}
+		}
+
+		params := &BatchCheckCommandParams{
+			AuthorizationModelID: ts.GetAuthorizationModelID(),
+			Checks:               checks,
+			StoreID:              ulid.Make().String(),
+		}
+
+		_, _, err := cmd.Execute(context.Background(), params)
+
+		var expectedErr *BatchCheckValidationError
```

Comment:
**Suggestion:** Make this more specific to the empty string error.

Reply:
Added an additional assertion below on the specific message. Or did you mean the error type itself?